pkg/policy/api/fqdn.go

```diff
@@ -76,6 +78,21 @@ func (s *FQDNSelector) sanitize() error {
 	return err
 }
 
+// ToRegex converts the given FQDNSelector to its corresponding regular
+// expression.
+func (s *FQDNSelector) ToRegex() *regexp.Regexp {
+	var preparedMatch string
+	if s.MatchName != "" {
+		preparedMatch = strings.ToLower(dns.Fqdn(s.MatchName))
```

Comment:
should this be `preparedMatch = "^"+preparedMatch+"$"` cc @raybejjani

Reply:
The call to `Validate` below should take care of the converting of this to a regular expression, as it calls `ToRegexp` in `pkg/fqdn/matchpattern/matchpattern.go`

Reply:
@ianvernon Please add a comment describing that. The `ToRegexp` also does `strings.ToLower` and this line is also doing `strings.ToLower`.

Reply:
I've added a comment to the function indicating that `MatchName` is ensured to be converted to a valid regular expression.